pkg/virtualKubelet/apiReflection/reflectors/incoming/pods.go

```diff
@@ -210,3 +210,27 @@ func (r *PodsIncomingReflector) isAllowed(ctx context.Context, obj interface{})
 	}
 	return !ok
 }
+
+// GetHomePod retrieves the shadow home pod given the foreign pod.
+var GetHomePod = func(pir *PodsIncomingReflector, foreignPod *corev1.Pod) (interface{}, error) {
```

Comment:
I feel this a bit weird as it is defined as a variable, instead of a proper function with the reflector as a receiver.
Still, I understand you need this for mocking during tests, so it seems fine to me.

Reply:
I have tried a different implementation that allows mocking the function. It might be prettier.

Reply:
Looks great!